modules/data.atmosphere/R/read.register.R

```diff
@@ -35,14 +35,18 @@ read.register <- function(register.xml, con) {
     } else if ((!is.null(register$format$id) & is.null(register$format$name)) 
                | 
                (!is.null(register$format$id) & is.null(register$format$mimetype))) {
-      register$format$name <- PEcAn.DB::db.query(
-        paste("SELECT name from formats where id = ", register$format$id), con)[[1]]
-      register$format$mimetype <- PEcAn.DB::db.query(
-        paste("SELECT mime_type from formats where id = ", register$format$id), con)[[1]]
+      # Retrieve format name and mimetype from the database
+      query.format.info <- PEcAn.DB::db.query(
+        paste("SELECT name, mimetype_id AS mimetype FROM formats WHERE id = ", register$format$id), con
+      )
+      
+      register$format$name <- query.format.info$name
+      register$format$mimetype <- query.format.info$mimetype
+
     } else if (is.null(register$format$id) & !is.null(register$format$name) & !is.null(register$format$mimetype)) {
       register$format$id <- PEcAn.DB::db.query(
         paste0("SELECT id from formats where name = '", register$format$name,
-               "' and mime_type = '", register$format$mimetype, "'"), con)[[1]]
+               "' and mimetype_id = '", register$format$mimetype, "'"), con)[[1]]
```

Comment:
I changed this below occurrence too to keep in check the Point 2 you mentioned.

Reply:
This won't work because mimetype_id is expecting a bigint (e.g. `1093`) while register$format$mimetype will be a string (e.g. "application/x-netcdf").

Reply:
```suggestion
               "' AND type_string = '", register$format$mimetype, "'"), con)[[1]]
```